src/app/shared/case-editor/case-edit-page.component.ts

```diff
@@ -71,15 +71,16 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
       .filter(caseField => !this.caseFieldService.isReadOnly(caseField))
       .every(caseField => {
         let theControl = this.editForm.controls['data'].get(caseField.id);
-        return this.checkDocumentField(caseField, theControl) && (theControl.valid || theControl.disabled);
+        return this.checkDocumentField(caseField, theControl)
+          && (this.caseFieldService.isOptional(caseField) || theControl.valid || theControl.disabled);
       });
   }
 
   private checkDocumentField(caseField: CaseField, theControl: AbstractControl): boolean {
```

Comment:
Can we extract the following code from **checkDocumentField** method:

`!(this.caseFieldService.isMandatory(caseField) && theControl === null)`

into a new method named **checkMandatoryField** and also extract the following block:

`(this.caseFieldService.isOptional(caseField) || theControl.valid || theControl.disabled)`

into a new method named  **checkOptionalField**?

Reply:
sure